packages/target-starknet-test/test/InputReturnTypes.test.ts

```diff
@@ -0,0 +1,123 @@
+import fs from 'fs'
+import { Contract, number, Provider } from 'starknet'
+const { toBN } = number
+import { expect } from 'earljs'
+
+import type { contract as _contract } from '../types/contract'
+
+describe('Type Transformation', () => {
+  let contract: _contract
+
+  before(async () => {
+    const provider = new Provider({ baseUrl: 'http://localhost:5000' })
+
+    async function deployContract(name: string, calldata: any[] = [], options: object = {}): Promise<Contract> {
+      const compiledContract = JSON.parse(fs.readFileSync(`./example-abis/${name}.json`).toString('ascii'))
+      const response = await provider.deployContract({
+        contract: compiledContract,
+        constructorCalldata: calldata,
+        ...options,
+      })
+      await provider.waitForTransaction(response.transaction_hash)
+      const address = response.address || ''
+      return new Contract(compiledContract.abi, address, provider)
+    }
+
+    contract = (await deployContract('contract')) as any as _contract
+  })
+
+  describe('Input Types', () => {
+    it('Parsing the felt in request', async () => {
+      await expect(contract.request_felt(3)).not.toBeRejected()
+    })
+
+    it('Parsing the array of felt in request', async () => {
+      await expect(contract.request_array_of_felts([1, 2])).not.toBeRejected()
+    })
+
+    it('Parsing the struct in request', async () => {
+      await expect(contract.request_struct({ x: 1, y: 2 })).not.toBeRejected()
+    })
+
+    it('Parsing the array of structs in request', async () => {
+      await expect(contract.request_array_of_structs([{ x: 1, y: 2 }])).not.toBeRejected()
+    })
+
+    it('Parsing the nested structs in request', async () => {
+      await expect(
+        contract.request_nested_structs({
+          p1: { x: 1, y: 2 },
+          p2: { x: 3, y: 4 },
+          extra: 5,
+        }),
+      ).not.toBeRejected()
+    })
+
+    it('Parsing the tuple in request', async () => {
+      await expect(contract.request_tuple([1, 2])).not.toBeRejected()
```

Comment:
i think we could drop `.not.toBeRejected ` everywhere here becuase if it would be rejected the whole test would bail 🤔

Reply:
It gives us slightly more civilised error messages. Without the huge stacktrace in case test fails.